src/pspm_interpolate.m

```diff
@@ -65,238 +51,169 @@
-  warning('ID:invalid_input', 'options.extrapolate must be numeric or logical');
-  return;
-end
-% 1.3 check data file argument
-% 1.3.1 define D
-if ischar(indata) || isstruct(indata) || isnumeric(indata)
-  D = {indata};
-elseif iscell(indata) ...
-    && sum(cellfun(@(f) isstruct(f), indata) | ...
-    cellfun(@(f) isnumeric(f), indata) | ...
-    cellfun(@(f) ischar(f), indata)) == numel(indata)
-  D = indata;
+
+% 1.3 determine the method
+if isnumeric(indata)
+    method = 1;
+elseif ischar(indata) && strcmpi(channel, 'all')
+    method = 2;
+elseif ischar(indata)
+    method = 3;
 else
-  warning('ID:invalid_input', ...
-    'Data must be either char, numeric, struct or cell array of char, numeric or struct.');
-  return;
+    warning('ID:invalid_input', 'Wrong input data format.');
+    return
 end
-% 1.3.2 define outdata
-if iscell(indata)
-  outdata = cell(size(D));
+
+% 1.4 User output
+if ischar(indata)
+    fprintf('\nInterpolating %s, ', indata); % user output
 else
-  outdata = {}; % initialise
+    fprintf('\nInterpolating ... '); % user output
 end
-%% 2 work on all data files
-for d = 1:numel(D)
-  % 2.1 load data
-  fn = D{d}; % determine file names
-  inline_flag = 0; % flag to decide what kind of data should be handled
-  if ischar(fn)
-    fprintf('\nInterpolating %s, ', fn); % user output
-  elseif isnumeric(fn)
-    inline_flag = 1;
-  end
-  if ~inline_flag % check and get datafile (not inline data must be loaded first)
-    if ischar(fn) && ~exist(fn, 'file')
-      warning('ID:nonexistent_file', 'The file ''%s'' does not exist.', fn);
-      outdata = {};
-      return;
-    end
-    % struct get checked if structure is okay; files get loaded
-    [lsts, infos, data] = pspm_load_data(fn, 0);
-    if any(lsts == -1)
-      warning('ID:invalid_data_structure', 'Cannot load data from data');
-      outdata = {};
-      break;
-    end
-    if numel(options.channel) > 0 && numel(options.channel{d}) > 0
-      % channel passed; try to get appropriate channel
-      work_channels = options.channel{d};
-      channel = data(work_channels);
-    else
-      % no channel passed; try to search appropriate channel
-      work_channels = find(cellfun(@(f) ~strcmpi(f.header.units, 'events'), data))';
-      channel = data(work_channels);
-    end
-    % sanity check channel should be a cell
-    if ~iscell(channel) && numel(channel) == 1
-      channel = {channel};
-    end
-    % look for event channel
-    ev = cellfun(@(f) strcmpi(f.header.units, 'events'), channel);
-    if any(ev)
-      warning('ID:invalid_chantype', 'Cannot interpolate event channel.');
-      return;
-    end
-  else
-    channel = {fn};
-  end
-  interp_frac = ones(numel(channel), 1);
-  for k = 1:numel(channel)
-    if inline_flag
-      dat = channel{k};
-    else
-      dat = channel{k}.data;
-    end
-    if numel(find(~isnan(dat))) < 2
+
+%% 2 work on data
+% 2.1 load data
+if method == 1
```

Comment:
I believe this `method` is different from `options.method` that is explained in help text. Maybe it is useful to explain the meaning of method 1, 2, 3 for future maintaining purposes.